packages/performance-tests/test/TransformerRegression.test.ts

```diff
@@ -95,16 +100,69 @@ const setupTestData = async () => {
 
 async function runRegressionTests() {
   const testIModels = await setupTestData();
-  let reporter = new Reporter();
+  const reporter = new Reporter();
   const reportPath = initOutputFile("report.csv", outputDir);
   const branchName =  await getBranchName();
 
   describe("Transformer Regression Tests", function () {
     testIModels.forEach(async (iModel) => {
+      let sourceDb: BriefcaseDb;
+      let reportInfo: ReporterInfo;
+
       describe(`Transforms of ${iModel.name}`, async () => {
+        before(async () => {
+          Logger.logInfo(loggerCategory, `processing iModel '${iModel.name}' of size '${iModel.tShirtSize.toUpperCase()}'`);
+          await iModel.getFileName();
+          assert(iModel._cachedFileName, "Imodel file name not Cached to TestIModel Object");
+          sourceDb = await BriefcaseDb.open({
+            fileName: iModel._cachedFileName,
+            readonly: true,
+          });
+          const fedGuidSaturation = sourceDb.withStatement(
+            `
+            SELECT
+            CAST(SUM(hasGuid) AS DOUBLE)/COUNT(*) ratio 
+            FROM (
+              SELECT IIF(FederationGuid IS NOT NULL, 1, 0) AS hasGuid,
+              1 AS total FROM bis.Element
+            )`,
+            (stmt) => {
+              assert(stmt.step() === DbResult.BE_SQLITE_ROW);
+              return stmt.getValue(0).getDouble();
+            }
+          );
+          Logger.logInfo(loggerCategory, `Federation Guid Saturation '${fedGuidSaturation}'`);
+          const toGb = (bytes: number) => `${(bytes / 1024 **3).toFixed(2)}Gb`;
+          const sizeInGb = toGb(fs.statSync(sourceDb.pathName).size);
+          Logger.logInfo(loggerCategory, `loaded (${sizeInGb})'`);
+          reportInfo = {
+            "Id": iModel.iModelId,
+            "T-shirt size": iModel.tShirtSize,
+            "Gb size": sizeInGb,
+            "Branch Name": branchName,
+            "Federation Guid Saturation 0-1": fedGuidSaturation,
+          };
+          sourceDb.close();
+        });
+
+        beforeEach(async () => {
+          assert(iModel._cachedFileName, "Imodel file name not Cached to TestIModel Object");
+          sourceDb = await BriefcaseDb.open({
+            fileName: iModel._cachedFileName,
```

Comment:
`_cachedFileName` should be private. Instead add a `let sourceFileName` variable and when you call `await iModel.getFileName()` in the `before` callback, instead do `sourceFileName = iModel.getFileName()` and just use that variable.

Reply:
done